daemon/workdir.go

```diff
@@ -0,0 +1,23 @@
+package daemon
+
+// ContainerCreateWorkdir creates the working directory. This is currently only used
+// on Windows to solve the issue arising from https://github.com/docker/docker/issues/27545,
+// which was initially fixed by https://github.com/docker/docker/pull/27884. But that fix
+// was too expensive in terms of performance to do consistently with Linux.
+// Instead, during a builder WORKDIR call on Windows, we come through this
+// function, avoiding a mount/unmount on every container create call.
+func (daemon *Daemon) ContainerCreateWorkdir(cID string) error {
+	container, err := daemon.GetContainer(cID)
+	if err != nil {
+		return err
+	}
+	err = daemon.Mount(container)
+	if err != nil {
+		return err
+	}
+	defer daemon.Unmount(container)
+	if err := container.SetupWorkingDirectory(0, 0); err != nil {
```

Comment:
Will be fixed in the forthcoming update. But I need some more work here to support Linux anyway for the UID/GID mapping calls for UserNS